exercises/tom_perkins/ex01_basics/src/counter.h

```diff
@@ -0,0 +1,31 @@
+#ifndef COUNTER_H_
+#define COUNTER_H_
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include <string>
+#include <map>
+#include <vector>
+#include <iostream>
+#include <algorithm>
+
+/**
+ * The Counter object keeps track of words that it has seen
+ * and can report on how many times it has seen each.
+ * Currently works for ASCII only (char not wchar_t).
+ */
+class Counter
```

Comment:
Use of separate well-named class for this is good. Not that it's wrong, but why did you use a class over say a free function?